src/components/TagForm/index.tsx

```diff
@@ -0,0 +1,53 @@
+'use client';
+
+import { TagSchemaType, tagSchema } from '@/lib/zod/tags';
+import { TextField, Button, Box } from '@mui/material';
+import { useForm } from 'react-hook-form';
+import { useRouter } from 'next/navigation';
+import { zodResolver } from '@hookform/resolvers/zod';
+import FormFieldError from '@/components/FormFieldError/FormFieldError';
+import { useMessage } from '../Providers/MessageProvider';
+import { post } from '@/lib/response/fetchUtil';
+
+export default function TagForm() {
+  const router = useRouter();
+  const { notify } = useMessage();
+
+  const {
+    register,
+    formState: { errors, isSubmitting },
+    handleSubmit,
+    reset,
+  } = useForm<TagSchemaType>({
+    resolver: zodResolver(tagSchema),
+  });
+
+  const submitForm = async (data: TagSchemaType) => {
+    const responseJson = await post('/api/tag', data);
```

Comment:
```suggestion
    data.name = data.name.replace(/\s{2,}/g, ' ').trim();
    const responseJson = await post('/api/tag', data);
```
You might want something like that to prevent duplicates better. The regex prevents adding for example "Agile Methods" and "Agile  Methods" (two spaces between there). Then it trims trailing and leading spaces to prevent stuff like " Docker", "Docker  " etc.

Reply:
Very good addition! Extra spaces can cause a lot of trouble; a managed to miss that.